controllers/cluster_controller.go

```diff
@@ -211,10 +214,33 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 	}
 
 	defer func() {
+		err := r.updateStatus(ctx, log, cluster)
+		if err != nil {
+			reterr = kerrors.NewAggregate([]error{reterr, err})
+		}
+
+		args := make([]string, 0, len(cluster.Status.Conditions))
+		for _, condition := range cluster.Status.Conditions {
+			args = append(args, fmt.Sprintf("(%s=%s %s)", condition.Type, condition.Status, condition.Reason))
+		}
+
+		log.Info("Current conditions", "conditions", strings.Join(args, ", "))
+
 		// Always attempt to patch the object and status after each reconciliation.
-		if err := patchHelper.Patch(ctx, cluster); err != nil {
+		patchOpts := []patch.Option{}
+
+		// Patch ObservedGeneration only if the reconciliation completed without error
```

Comment:
I like this
But I would extend the comment to explain the "why", here I'm only getting the "what" (which I could possible get from just reading the code)